tests/Resources/ti.database.test.js

```diff
@@ -474,6 +474,44 @@ describe('Titanium.Database', function () {
 			});
 		});
 
+		it('returns a Promise', function (finish) {
+			this.timeout(5000);
+			const db = Ti.Database.open('execute_async.db');
+			// Execute a query to create a test table
+			const result = db.executeAsync('CREATE TABLE IF NOT EXISTS testTable (text TEXT, number INTEGER)');
+			result.should.be.a.Promise();
+			// Delete any existing data if the table already existed
+			result.then(() => db.executeAsync('DELETE FROM testTable'))
+				.then(() => {
+					// Define test data
+					const testName = 'John Smith';
+					const testNumber = 123456789;
+
+					// Insert test data into the table
+					return db.executeAsync('INSERT INTO testTable (text, number) VALUES (?, ?)', testName, testNumber);
+				}).then(() => {
+					// Validate that only one row has been affected
+					should(db.rowsAffected).be.eql(1);
+
+					// Execute a query to return the rows of the database
+					return db.executeAsync('SELECT rowid, text, number FROM testTable');
+				}).then(rows => {
+					try {
+						// Validate the returned 'rows' object
+						should(rows).be.a.Object();
+						should(rows.rowCount).be.eql(1);
+						should(rows.fieldCount).be.eql(3);
+						should(rows.validRow).be.true();
+					} finally {
```

Comment:
<!--
  0 failure: 
  2 warning:  tests/Resources/t..., tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.database.test.js;
  Line: 505;
-->

- :warning: tests/Resources/ti.database.test.js line 505 – Each then() should return a value or throw (promise/always-return)
- :warning: tests/Resources/ti.database.test.js line 505 – Each then() should return a value or throw (promise/always-return)